client/src/controllers/SwapController.ts

```diff
@@ -279,15 +292,16 @@ export class SwapController extends Controller<
     }) as CustomEvent<{ requestUrl: string }>;
 
     if (beginEvent.defaultPrevented) return Promise.resolve();
-    const formMethod = this.formElement.getAttribute('method') || undefined;
+    const formMethod =
+      this.formElement.getAttribute('method')?.toUpperCase() || 'GET';
     return fetch(requestUrl, {
       headers: {
         'x-requested-with': 'XMLHttpRequest',
         [WAGTAIL_CONFIG.CSRF_HEADER_NAME]: WAGTAIL_CONFIG.CSRF_TOKEN,
       },
       signal,
       method: formMethod,
-      body: formMethod !== 'get' ? data : undefined,
+      body: formMethod !== 'GET' ? data : undefined,
```

Comment:
I decided to normalise the method name to upper case instead of lower case, because in the context of the `fetch()` spec (and not the HTML `<form>` spec), it says:

> To normalize a [method](https://fetch.spec.whatwg.org/#concept-method), if it is a [byte-case-insensitive](https://infra.spec.whatwg.org/#byte-case-insensitive) match for `DELETE`, `GET`, `HEAD`, `OPTIONS`, `POST`, or `PUT`, [byte-uppercase](https://infra.spec.whatwg.org/#byte-uppercase) it.

https://fetch.spec.whatwg.org/#methods